nearform/mercurius-federation-info-graphiql-plugin
# Add input for deep searching the services's info

This ticket closes https://github.com/nearform/mercurius-federation-info-graphiql-plugin/issues/85

In this PR is introduced the ability to search through services's information tree and takes into consideration: 
- debouncing the search 
- case insensitivity
- dark theme ui
- updating service count on filtering


**What is not part of this PR** (will be handled in https://github.com/nearform/mercurius-federation-info-graphiql-plugin/issues/120)
- Save search while navigating between tabs
- Expand data (TBD solution) https://nearform.slack.com/archives/C045D7SPLD8/p1671186046014239
- Remove recursive search?


<img width="782" alt="Screenshot 2022-12-19 at 15 12 36" src="https://user-images.githubusercontent.com/3214357/208448891-bc9daf00-c6cc-4295-80c2-e31d24690978.png">
<img width="319" alt="Screenshot 2022-12-19 at 15 31 45" src="https://user-images.githubusercontent.com/3214357/208449144-f8fab41b-0dd9-4178-a732-d866dc6fa227.png">
<img width="780" alt="Screenshot 2022-12-19 at 15 12 49" src="https://user-images.githubusercontent.com/3214357/208448896-f923f275-a021-4ff2-b3e0-4384c16a50b4.png">


To test this PR:
- install the dependencies with npm ci
- execute the command npm run example
- open your browser to http://localhost:3001/graphiql
- click on the plugin tab
- click on search input any start typing the name
- matching services name should be filtered out (type names are not taken into account)
